### PR TITLE
Alerting: Fix go template parsing

### PR DESCRIPTION
**Pull request by konrad147**
**What is this feature?**
This PR fixes templates parsing when a template ends with multiple `{{ end }}` blocks

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

**Review by soniaAguilarPeiron — APPROVED** [review]
LGTM!!

**Review by tomratcliffe — COMMENTED** [review]
:+1: We could also test this by adding a mock template to `mocks/server/entities/alertmanager-config/grafana-alertmanager-config.ts` which has the multiple end blocks, and checking that the page renders appropriately?

**Comment by konrad147** [comment]
>We could also test this by adding a mock template to mocks/server/entities/alertmanager-config/grafana-alertmanager-config.ts which has the multiple end blocks, and checking that the page renders appropriately?

We could, but I believe a simple unit test would be a better fit for cases like this. It allows us to cover a wide range of sample templates without the need to set up the entire UI.

File: public/app/features/alerting/unified/components/receivers/form/fields/utils.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ import { Template } from './TemplateSelector';
 export function parseTemplates(templatesString: string): Template[] {
   const templates: Record<string, Template> = {};
   const stack: Array<{ type: string; startIndex: number; name?: string }> = [];
-  const regex = /{{(-?\s*)(define|end|if|range|else|with|template)(\s*.*?)?(-?\s*)}}/gs;
+  const regex = /{{\s*(-?)\s*(define|end|if|range|else|with|template|block)\b(.*?)\s*(-?)\s*}}/gs;
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/receivers/form/fields/utils.ts:23]: The start delimiter seems incorrect as that would match `{{ -` but the syntax only allows `{{` or `{{-`. Dito for the end delimiter.

**Review by gillesdemey — APPROVED** [review]

**Comment by DorisJeanParish** [comment]
Apple.id199354757 Appledatabase.com ID:049921093

**Pull request merged**